python/pyspark/sql/types.py

```diff
@@ -1448,6 +1448,54 @@ def __repr__(self):
         else:
             return "<Row(%s)>" % ", ".join(self)
 
+    def __eq__(self, other):
+        """
+        Test for equality with `other`.
+
+        :param other: other Row for comparison
+
+        >>> Row(name="Alice", age=11) == Row(age=11, name="Alice")
+        True
+        >>> R1 = Row('a', 'b')
+        >>> R2 = Row('b', 'a')
+        >>> R1(1, 2) == R2(2, 1)
```

Comment:
Although, carrying that to its logical conclusion, then `R1 == R2` should return `True` in this case, which it currently doesn't.